app/services/make_an_offer.rb

```diff
@@ -8,14 +8,16 @@ class MakeAnOffer
 
   validate :validate_offer_conditions
 
-  def initialize(application_choice:, offer_conditions: nil)
+  def initialize(application_choice:, offer_conditions: nil, offer_course_option_id: application_choice.course_option.id)
     @application_choice = application_choice
     @offer_conditions = offer_conditions
+    @offer_course_option_id = offer_course_option_id
   end
 
   def save
     return unless valid?
 
+    application_choice.update!(offer_course_option_id: @offer_course_option_id)
```

Comment:
Could move down to `application_choice.offer` so that it saves in the same query?

Reply:
I thought about doing this; I'll go for it then.

When we were kicking this off, we mentioned querying to measure which choices have had an alternate offer come in from a provider. We can store that information inside of the JSON blob, it will just make it slightly trickier to retrieve. That being said, it's not impossible, so let's try it.

Reply:
Sorry I meant something different! Having it as a FK like here is definitely the way to go.

I just meant:

```diff
- application_choice.update!(offer_course_option_id: @offer_course_option_id)
ApplicationStateChange.new(application_choice).make_offer!
+ application_choice.offer_course_option_id = @offer_course_option_id
application_choice.offer = { 'conditions' => (@offer_conditions || []) }

application_choice.offered_at = Time.now
application_choice.save!
```

Reply:
Ah yes, not using `update!` because we `save!` later anyway, doh! 👍